File: dayFour/python/account_functions_main.py

from account_functions import *
      
balance = 0.0
balance = checkBalance(balance)
newBalance = 0.0
amount = 0.0
transactions = ""
options = 0
while options != 4 :

    MenuFunctions = """
        [1] Deposit Money
        [2] Withdraw Money
        [3] Show transactions history
        [4] Exit the program 
    """
    print(MenuFunctions)
    
    options = int(input("Enter your choice: "))
    
    
    match options :
        case 1 : 
            amount = float(input("Enter deposit amount: ₦"))
            newBalance = deposit(amount, balance, "")
            print(f"Withdrew: ₦{amount} | New Balance: ₦{newBalance}")
            store_transaction(amount, balance)
        
        case 2 : 
            withdraw = float(input("Enter withdrawal amount: ₦"))
            balance = newBalance
            if amount > balance :
                print("Withdrawal failed: insufficent funds")
                break
                
            newBalance = withdraw(amount, balance, "")
            print(f"Withdrew: ₦{amount} | New Balance: ₦{newBalance}")

        
#        case 3 : 
            
        
        case 4 : 
            balance = newBalance
            print(f"Final Balance: ₦{balance}")
            print("Thank you for using Transaction Log App!")
            break

    
       

File: dayFour/java/AccountFunctionsMain.java
import java.util.Scanner;
public class AccountFunctionsMain{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        
        double balance = 0.0;
        balance = AccountFunctions.checkBalance(balance);
        double newBalance = 0.0;
        double amount = 0.0;
        String transactions = "";
        int options = 0;
        while(options != 4){

            String MenuFunctions = """
                [1] Deposit Money
                [2] Withdraw Money
                [3] Show transactions history
                [4] Exit the program 
            """;
            System.out.println(MenuFunctions);
            
            System.out.print("Enter your choice: ");
            options = input.nextInt();
            
            switch(options){
                case 1 : {
                    System.out.print("Enter deposit amount: ₦");
                    amount = input.nextDouble();
                    newBalance = AccountFunctions.deposit(amount, balance, "");
                    System.out.printf("Deposited: ₦%.2f | New Balance: ₦%.2f%n", amount, newBalance);
                    break;
                }
                case 2 : {
                    System.out.print("Enter withdrawal amount: ₦");
                    amount = input.nextDouble();
                    balance = newBalance;
                        if (amount > balance) {
                            System.out.println("Withdrawal failed: insufficent funds");
                            break;
                        }
                    newBalance = AccountFunctions.withdraw(amount, balance, "");
                    System.out.printf("Withdrew: ₦%.2f | New Balance: ₦%.2f%n", amount, newBalance);
                    break;
                }
                case 3 : {
                    
                    break;
                }
                case 4 : {
                    balance = newBalance;
                    System.out.printf("Final Balance: ₦%.2f%n", balance);
                    System.out.println("Thank you for using Transaction Log App!");
                    break;
                }
            }
       }
            
   }
            
}

